core/DataTable.php

```diff
@@ -573,7 +573,13 @@ public function addDataTable(DataTable $tableToSum)
             $this->aggregateRowFromSimpleTable($row);
         } else {
             $columnAggregationOps = $this->getMetadata(self::COLUMN_AGGREGATION_OPS_METADATA_NAME);
-            foreach ($tableToSum->getRows() as $row) {
+            foreach ($tableToSum->getRowsWithoutSummaryRow() as $row) {
+                $this->aggregateRowWithLabel($row, $columnAggregationOps);
+            }
+            // we do not use getRows() as this method might get called 100k times when aggregating many datatables and
+            // this takes a lot of time.
+            $row = $tableToSum->getRowFromId(DataTable::ID_SUMMARY_ROW);
+            if ($row) {
```

Comment:
Not strictly related to this change, maybe it would increase code clarity if DataTable::$summaryRow were stored in DataTable::$rows? I haven't looked to see if this would be do-able or not, maybe something to think about in another pull request.